docs/troubleshooting.md

```diff
@@ -28,14 +34,159 @@ Please clear your Azure profile directory with `rm -rf ~/.azure` and retry after
+
+### ExternalDNS is not updating DNS zone records
+
+The DNS host records (__A__) can be listed using the following command:
+
+On Google cloud:
+
+```bash
+gcloud dns record-sets list --zone $(gcloud dns managed-zones list --filter dnsName:${BKPR_DNS_ZONE} --format='value(name)') --filter type=A
+```
+
+On Azure cloud:
+
+```bash
+az network dns record-set list --resource-group ${AZURE_RESOURCE_GROUP} --zone-name ${BKPR_DNS_ZONE} --query "[?arecords!=null]" --output table
+```
+
+BKPR, by default, creates host records for Prometheus, Grafana and Kibana dashboards. These records should be listed in the output of the above command.
+
+__Troubleshooting__:
+
+ExternalDNS automatically manages host records for Ingress resources in the cluster. When a Ingress resource is created it could take a few minutes for it to be seen by ExternalDNS.
+
+If the records are not updated a few minutes after the Ingress resource is created, use the following command to inspect the container logs of the `external-dns` Pod to discover any error conditions that may have been encountered.
+
+```bash
+kubectl -n kubeprod logs $(kubectl -n kubeprod get pods -l name=external-dns -o name)
+```
+
+### DNS glue records are not configured
+
+The DNS glue record setup is the most basic requirement for the DNS address resolution to work correctly. The glue records are typically configured at the DNS domain registrar.
+
+Use the following command to query the glue record configuration for your domain:
+
+```bash
+dig ${BKPR_DNS_ZONE} NS +noall +answer
+```
+
+The command should list one or more Nameserver (__NS__) records configured for your domain.
+
+__Troubleshooting__:
+
+If the above listed command does not return any __NS__ records, it indicates that the DNS glue records have not been configured at your domain registrar. At the same time, it is also possible the glue records are configured with wrong values.
+
+First, use the following command to query the values of the NS records that should be set up as glue records.
+
+On Google cloud,
+
+```bash
+gcloud dns managed-zones describe \
+  $(gcloud dns managed-zones list --filter dnsName:${BKPR_DNS_ZONE} --format='value(name)')
+```
+
+On Azure cloud,
+
+```bash
+az network dns zone show \
+  --name ${BKPR_DNS_ZONE} \
+  --resource-group ${AZURE_RESOURCE_GROUP} \
+  --query nameServers \
+  --output table
+```
+
+Next, use your domain registrar's portal to add __NS__ records for your domain with the values displayed in output of the previous command.
+
+### DNS propagation has not completed
+
+Nameserver changes can typically take 0 to 24 hours to take effect, but they are known to take as long as 48 hours to go into full effect. This is because it takes time for the DNS to take effect across the internet. The actual time of propagation may vary based on your network setup.
```

Comment:
I would only say "Nameserver changes typically take 0 to 24 hours to take effect, but they may also take as long as 48 hours."